code/modules/mob/living/simple_animal/friendly/diona.dm

```diff
@@ -162,28 +159,32 @@
 		unEquip(W)
 
 	qdel(src)
+	return TRUE
 
-/mob/living/simple_animal/diona/verb/steal_blood()
-	set category = "Diona"
-	set name = "Steal Blood"
-	set desc = "Take a blood sample from a suitable donor."
-
+/mob/living/simple_animal/diona/proc/steal_blood()  //called by the hud (/obj/screen/diona/blood)
+	if(!isnymph(src) || src.stat)
+		return FALSE
+	
 	var/list/choices = list()
 	for(var/mob/living/carbon/human/H in oview(1,src))
-		choices += H
+		if(Adjacent(H))
```

Comment:
may want to add a check on their species datum that they have blood suitable for a diona